### ISSUE TITLE
Implement connection state freshness check

### ISSUE THREAD

**Issue by SimonWoolf** [ably/ably-ios#645]
Spec PR: https://github.com/ably/docs/pull/331/files

This is quite high priority; see https://github.com/ably/realtime/issues/1289#issuecomment-341152556 for why, or ask me or Matt in #client-libraries or slack

**Comment by funkyboy** [comment]
@SimonWoolf This looks like it's been "fixed" here: https://github.com/ably/ably-ios/pull/660 
Can I close?

**Comment by SimonWoolf** [comment]
> This looks like it's been "fixed" here: #660

No, that pr didn't implement RTN15g, which is what this issue is about. See comment https://github.com/ably/ably-ios/pull/660#issuecomment-350664088 for more context.
